### PR TITLE
cpython: Fixed build error when sqlite3 wasn't used.

### LINKED ISSUE #9890: [package] cpython/3.10.0: build fails if SQLite3 is disabled
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **cpython/3.10.0**
 
### Steps to reproduce (Include if Applicable)
If you try to build "cpython" with SQLite3 disabled you'll get a build error from Conan, as it tries to access the "self.options['sqlite3']" structure which only exists if SQLite3 is enabled.

It's a simple fix, PR incoming.

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
ERROR: cpython/3.10.0: Error in build() method, line 427
	autotools = self._configure_autotools()
while calling '_configure_autotools', line 234
	"--enable-loadable-sqlite-extensions={}".format(yes_no(not self.options["sqlite3"].omit_load_extension)),
	ConanException: option 'omit_load_extension' doesn't exist
```

</details>

### PR DESCRIPTION
**Pull request by erikogenvik**
Specify library name and version:  **cpython/3.10.0**

If SQLite3 isn't used then the build will fail, since it tries to access the ```self.options["sqlite3"]``` option, which only exists if SQLite3 is enabled. This PR fixes that.

fixes #9890

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by Croydon — APPROVED** [review]

**Review by uilianries — DISMISSED** [review]
LGTM

**Comment by uilianries** [comment]
@erikogenvik please, ask for user access on #4. The CI bot only runs approved users.

**Comment by erikogenvik** [comment]
This seems to fail to build because of an issue unrelated to the actual PR. Or am I missing something?

**Comment by uilianries** [comment]
Correct. It was not able to build with apple-clang 11

**Comment by erikogenvik** [comment]
I don't know how to proceed with this PR. The actual code change is only four lines. But after merging with master it now fails with linter errors, which are unrelated to the changes.

**Review by jwillikers — APPROVED** [review]

**Comment by uilianries** [comment]
@erikogenvik It's because CCI is migrating to support Conan 2.0. According to the [Roadmap](https://github.com/conan-io/conan-center-index/blob/master/docs/v2_roadmap.md), all recipes will need to be adapted. Unfortunately, you reached an outdated recipe, and now, it should be updated, otherwise, the CI will not approve. It's the same rule for any recipe, even if you didn't touch those imports.

I gonna open a PR fixing those points for you .

**Comment by uilianries** [comment]
@erikogenvik Please, consider https://github.com/erikogenvik/conan-center-index/pull/1

**Comment by erikogenvik** [comment]
> @erikogenvik Please, consider [erikogenvik#1](https://github.com/erikogenvik/conan-center-index/pull/1)

Thanks, merged just now.

File: recipes/cpython/all/conanfile.py
Comment (jwillikers) [inline recipes/cpython/all/conanfile.py:158]: Would it be possible to use `check_min_vs()` here so that it works with the newer `msvc` compiler?

File: recipes/cpython/all/conanfile.py
Comment (jwillikers) [inline recipes/cpython/all/conanfile.py:180]:
```suggestion
               and (not is_msvc(self) or Version(self._version_number_only) >= "3.8")
```

File: recipes/cpython/all/conanfile.py
Comment (jwillikers) [inline recipes/cpython/all/conanfile.py:733]:
```suggestion
        pythonhome_required = is_msvc(self) or is_apple_os(self)
```

**Review by jwillikers — CHANGES_REQUESTED** [review]
1 / 1 files viewed
Finish your review

 
Add heading textAdd bold text, <Ctrl+b>Add italic text, <Ctrl+i>
Add a quote, <Ctrl+Shift+.>Add code, <Ctrl+e>Add a link, <Ctrl+k>
Add a bulleted list, <Ctrl+Shift+8>Add a numbered list, <Ctrl+Shift+7>Add a task list, <Ctrl+Shift+l>
Directly mention a user or team
Reference an issue, pull request, or discussion
Add saved reply
Leave a comment
No file chosen
Attach files by dragging & dropping, selecting or pasting them.
Styling with Markdown is supported
Comment
Submit general feedback without explicit approval.

Approve
Submit feedback approving these changes.

Request changes
Submit feedback suggesting changes.

3 pending comments

File: recipes/cpython/all/conanfile.py
Comment (erikogenvik) [reply recipes/cpython/all/conanfile.py:158]: These are changes suggested by @uilianries , I'll let him handle them.

Note that this PR initially only suggested a one line change, as specified in this commit: https://github.com/conan-io/conan-center-index/pull/9891/commits/d6103f2faaab4f4b92ca32519195872502043e37

Comment (uilianries) [reply recipes/cpython/all/conanfile.py:158]: `check_min_vs()` raises `ConanInvalidConfiguration`. Here, we just want an warning message. Docs: https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#check-min-vs

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — COMMENTED** [review]
There's two comments for "msvc" compiler support but I am not sure those are enough.

I am giving this a check, great work everyone!

**Review by prince-chrismc — DISMISSED** [review]

**Comment by erikogenvik** [comment]
It's failing when building for MacOS, and it's not clear to me why.

**Review by jwillikers — APPROVED** [review]

**Comment by jwillikers** [comment]
@erikogenvik Possibly related to python/cpython#82624?

File: recipes/cpython/all/conanfile.py
Comment (prince-chrismc) [inline recipes/cpython/all/conanfile.py:180]: Missing the import. And looking at the code there's a bunch h of places its needed

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request closed without merging**
